nncf/onnx/quantization/quantize.py

```diff
@@ -0,0 +1,117 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Optional
+
+import numpy as np
+import onnx
+
+from nncf import Dataset, TargetDevice
+from nncf.common.quantization.structs import QuantizationPreset
+from nncf.experimental.onnx.tensor import ONNXNNCFTensor
+from nncf.experimental.post_training.algorithms.quantization import PostTrainingQuantization
+from nncf.experimental.post_training.algorithms.quantization import PostTrainingQuantizationParameters
+from nncf.experimental.post_training.api.dataset import Dataset as PTQDataset
+from nncf.experimental.post_training.compression_builder import CompressionBuilder
+
+
+# TODO(alexsu52): It is a workaround and should be removed.
```

Comment:
It would be great to have more information in the comment that explains the reason because I was about to place a comment that we don't need this object)

Reply:
The short answer, ONNX expects the dataset instance with following implemented interfaces https://github.com/openvinotoolkit/nncf/blob/develop/nncf/experimental/post_training/api/dataset.py

Reply:
Why not have exactly this text in the in-code comment then?

Reply:
Done